packages/jest-core/src/TestScheduler.ts

```diff
@@ -177,32 +187,73 @@ export default class TestScheduler {
     });
 
     const testRunners: {[key: string]: TestRunner} = Object.create(null);
-    contexts.forEach(({config}) => {
+    const contextsByTestRunner = new WeakMap<TestRunner, Context>();
+    contexts.forEach(context => {
+      const {config} = context;
       if (!testRunners[config.runner]) {
         const Runner: typeof TestRunner = require(config.runner);
-        testRunners[config.runner] = new Runner(this._globalConfig, {
+        const runner = new Runner(this._globalConfig, {
           changedFiles: this._context?.changedFiles,
           sourcesRelatedToTestsInChangedFiles: this._context
             ?.sourcesRelatedToTestsInChangedFiles,
         });
+        testRunners[config.runner] = runner;
+        contextsByTestRunner.set(runner, context);
       }
     });
 
-    const testsByRunner = this._partitionTests(testRunners, tests);
+    const testsByRunner = TestScheduler._partitionTests(testRunners, tests);
```

Comment:
change this back to `this` should fix build

Reply:
Ah I missed this, thanks!